drivers/input/touchscreen/synaptics_dsx_i2c.c

```diff
@@ -3809,11 +3857,14 @@ static void synaptics_rmi4_set_params(struct synaptics_rmi4_data *rmi4_data)
 	set_bit(KEY_HOMEPAGE, rmi4_data->input_dev->keybit);
 	set_bit(KEY_F3, rmi4_data->input_dev->keybit);
 	set_bit(KEY_DOUBLE_TAP, rmi4_data->input_dev->keybit);
-	set_bit(KEY_GESTURE_CIRCLE, rmi4_data->input_dev->keybit);
-	set_bit(KEY_GESTURE_SWIPE_DOWN, rmi4_data->input_dev->keybit);
-	set_bit(KEY_GESTURE_V, rmi4_data->input_dev->keybit);
-	set_bit(KEY_GESTURE_LTR, rmi4_data->input_dev->keybit);
-	set_bit(KEY_GESTURE_GTR, rmi4_data->input_dev->keybit);
+	set_bit(EV_MSC, rmi4_data->input_dev->evbit);
+	rmi4_data->input_dev->mscbit[0] = BIT_MASK(MSC_GESTURE);
```

Comment:
Sorry, I think I was confused for a while.

With here is only one gesture "event" (that can have multiple values). So unless there are other MSC_xxx events, then your original code was correct.

If there are other than MSC_GESTURE events, then using:

    set_bit(MSC_GESTURE, rmi4_data->input_dev->mscbit);

would not clear other already set bits.